plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/GcpLocalRunTab.java

```diff
@@ -200,14 +203,16 @@ public void modifyText(ModifyEvent event) {
     });
     Button browse = new Button(composite, SWT.NONE);
     browse.setText(Messages.getString("button.browse")); //$NON-NLS-1$
+    browse.addSelectionListener(new FileFieldSetter(serviceKeyInput,
+        new String[] {"*.json", "*"})); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
why * AND *.json? Why not just *.json?

Also this might be a little easier to read with an extr local variable for the file field setter.

Reply:
If we can be certain the file should always be `.json`, we can git rid of `*`. I put `*` as a fallback because otherwise, there is no way users can select a non-`.json` file. (They could still enter anything in the input field directly though.)

Reply:
Pantheon will always create a .json file. I think that's good enough.